### PR TITLE
fix: [PIE-6077]: amazon machine image artifact

### PR DESCRIPTION
**Pull request by satyam-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonMachineImage/AmazonMachineImage.tsx [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+    //   spec: {
+    //     connectorRef: connectorId,
+    //     artifactPath: formData.spec.artifactPath,
+    //     build: formData.spec.build,
+    //     jobName:
+    //       getMultiTypeFromValue(formData.spec?.jobName) === MultiTypeInputType.FIXED
+    //         ? (formData.spec?.jobName as SelectOption).label
+    //         : formData.spec?.jobName
+    //   }
+    // })
```
Comment (ishantmehtaharness) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonMachineImage/AmazonMachineImage.tsx:298]: we can remove commented code

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonMachineImage/AmazonMachineImage.tsx [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+export function AmazonMachineImage(
+  props: StepProps<ConnectorConfigDTO> & ImagePathProps<AmazonMachineImageInitialValuesType>
+): React.ReactElement {
+  const { getString } = useStrings()
+  const { context, handleSubmit, initialValues, prevStepData, artifactIdentifiers } = props
+  const isIdentifierAllowed = context === ModalViewFor.SIDECAR || !!props.isMultiArtifactSource
+  const isTemplateContext = context === ModalViewFor.Template
+
+  const getInitialValues = (): AmazonMachineImageInitialValuesType => {
+    console.log('initialValues', initialValues)
```
Comment (ishantmehtaharness) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonMachineImage/AmazonMachineImage.tsx:280]: console log?

**Comment by satyam-harness** [comment]
retrigger titlecheck

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonMachineImage/AmazonMachineImage.tsx (hunk 1 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonMachineImage/AmazonMachineImage.tsx]: removed

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonMachineImage/AmazonMachineImage.tsx (hunk 2 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonMachineImage/AmazonMachineImage.tsx]: PR is under test, added that for dev testing, will remove while merging

**Review by vkbansal-harness — APPROVED** [review]

**Pull request merged**